homeassistant/components/tasmota_irhvac/climate.py

```diff
@@ -0,0 +1,748 @@
+        await self.hass.async_add_executor_job(self.send_ir)
+        await self.async_update_ha_state()
+
+    @property
+    def min_temp(self):
+        """Return the minimum temperature."""
+        if self._min_temp:
+            return self._min_temp
+
+        # get default temp from super class
+        return super().min_temp
+
+    @property
+    def max_temp(self):
+        """Return the maximum temperature."""
+        if self._max_temp:
+            return self._max_temp
+
+        # Get default temp from super class
+        return super().max_temp
+
+    async def _async_sensor_changed(self, entity_id, old_state, new_state):
+        """Handle temperature changes."""
+        if new_state is None:
+            return
+
+        self._async_update_temp(new_state)
+        await self.async_update_ha_state()
+
+    @callback
+    def _async_update_temp(self, state):
+        """Update thermostat with latest state from sensor."""
+        try:
+            self._cur_temp = float(state.state)
+        except ValueError as ex:
+            _LOGGER.error("Unable to update from sensor: %s", ex)
+
+    @property
+    def _is_device_active(self):
+        """If the toggleable device is currently active."""
+        return self.power_mode == STATE_ON
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        return self._support_flags
+
+    async def async_set_preset_mode(self, preset_mode):
+        """Set new preset mode.
+
+        This method must be run in the event loop and returns a coroutine.
+        """
+        if preset_mode == PRESET_AWAY and not self._is_away:
+            self._is_away = True
+            self._saved_target_temp = self._target_temp
+            self._target_temp = self._away_temp
+        elif preset_mode == PRESET_NONE and self._is_away:
+            self._is_away = False
+            self._target_temp = self._saved_target_temp
+        await self.hass.async_add_executor_job(self.send_ir)
+        await self.async_update_ha_state()
+
+    def send_ir(self):
+        """Send the payload to tasmota mqtt topic."""
+        curr_operation = self._hvac_mode
+        fan_speed = self.fan_mode
+        # tweek for some ELECTRA_AC devices
+        if HVAC_FAN_MAX_HIGH in self._fan_list and HVAC_FAN_AUTO_MAX in self._fan_list:
+            if self.fan_mode == FAN_HIGH:
+                fan_speed = HVAC_FAN_MAX
+            if self.fan_mode == HVAC_FAN_MAX:
+                fan_speed = HVAC_FAN_AUTO
+
+        # Set the swing mode - default off
+        swing_h = STATE_OFF
+        swing_v = STATE_OFF
+        if self.swing_mode == SWING_BOTH:
+            swing_h = STATE_AUTO
+            swing_v = STATE_AUTO
+        elif self.swing_mode == SWING_HORIZONTAL:
+            swing_h = STATE_AUTO
+        elif self.swing_mode == SWING_VERTICAL:
+            swing_v = STATE_AUTO
+        # Poplate the payload
+        payload = (
+            '{"Vendor":"'
```

Comment:
Use f-strings instead of string concatenation, see the [style guidelines](https://developers.home-assistant.io/docs/en/next/development_guidelines.html#use-new-style-string-formatting) for more information.

Reply:
Done!